mlx/coverity_services.py

```diff
@@ -1,614 +1,396 @@
+                    ((column["name"], column["columnKey"]) for column in columns)
+                )
+        return self.columns
+
+    def retrieve_checkers(self):
+        """Retrieve the list of checkers from the server.
 
-        # apply any filter on impact status
-        if filters['impact']:
-            self.handle_attribute_filter(filters['impact'], 'Impact', IMPACT_LIST, filter_spec.impactNameList)
+        Returns:
+            list[str]: The list of valid checkers
+        """
+        if not self.checkers:
+            url = f"{self.api_endpoint}/checkerAttributes/checker"
+            checkers = self._request(url)
+            if checkers and "checkerAttributedata" in checkers:
+                self._checkers = [checker["key"] for checker in checkers["checkerAttributedata"]]
+        return self.checkers
 
-        # apply any filter on issue kind
-        if filters['kind']:
-            self.handle_attribute_filter(filters['kind'], 'Kind', KIND_LIST, filter_spec.issueKindList)
+    def _request(self, url, data=None):
+        """Perform a POST or GET request to the specified url.
+        Uses a GET request when data is `None`, uses a POST request otherwise
 
-        # apply any filter on classification
-        if filters['classification']:
-            self.handle_attribute_filter(filters['classification'],
-                                         'Classification',
-                                         CLASSIFICATION_LIST,
-                                         filter_spec.classificationNameList)
+        Args:
+            url (str): The URL for the request
+            data (dict): Optional data to send
 
-        # apply any filter on action
-        if filters['action']:
-            self.handle_attribute_filter(filters['action'], 'Action', ACTION_LIST, filter_spec.actionNameList)
+        Returns:
+            dict: the content of server's response
 
-        # apply any filter on Components
-        if filters['component']:
-            self.handle_component_filter(filters['component'], filter_spec)
+        Raises:
+            requests.HTTPError
+        """
+        if data:
+            response = self.session.post(url, json=data)
+        else:
+            response = self.session.get(url)
+        if response.ok:
+            return response.json()
+        try:
+            err_msg = response.json()["message"]
+        except (requests.exceptions.JSONDecodeError, KeyError):
+            err_msg = response.content.decode()
+        self.logger.warning(err_msg)
+        return response.raise_for_status()
 
-        # apply any filter on CWE values
-        if filters['cwe']:
-            self.handle_attribute_filter(filters['cwe'], 'CWE', None, filter_spec.cweList)
+    @staticmethod
+    def validate_filter_option(name, req_csv, valid_attributes, allow_regex=False):
+        """Add filter when the attribute is valid. If `valid_attributes` is empty or falsy,
```

Comment:
```suggestion
        """Add filter when the attribute is valid. If `valid_attributes` is empty,
```

Reply:
Resolved in db357d59c14d2fe8ec644fa8e55117598ef89a7a